apis/r/R/TileDBArray.R

```diff
@@ -49,6 +49,29 @@ TileDBArray <- R6::R6Class(
       invisible(self)
     },
 
+    #' @description Close and reopen the TileDB object in a new mode
+    #'
+    #' @param mode New mode to open the object in; choose from:
+    #' \itemize{
+    #'  \item \dQuote{\code{READ}}
+    #'  \item \dQuote{\code{WRITE}}
+    #' }
+    #' By default, reopens in the opposite mode of the current mode
+    #'
+    #' @return Invisibly returns \code{self}
+    #'
+    reopen = function(mode = NULL) {
+      modes <- c(READ = 'WRITE', WRITE = 'READ')
+      oldmode <- self$mode()
+      mode <- mode %||% modes[oldmode]
```

Comment:
The flipping read/write is clever but maybe too hard too read?   

Also what happens if mode were "RESUME" ?

Reply:
Per conversation with @mojaveazure `mode` and `ingest_mode` are distinct in Python, and must be in R as well.

Reply:
For `$open()` (and therefore `$reopen()`); mode cannot be `"RESUME"`

On the Puython side, resume-mode exists only in `tiledbsoma.io`; for parity, this means that resume-mode exists only in the `write_soma()` methods and is handled there, but the mode of the array itself cannot be `"RESUME"`